client/lib/cart/store/index.js

```diff
@@ -151,6 +165,47 @@ CartStore.dispatchToken = Dispatcher.register( payload => {
 		case CART_ITEM_REPLACE:
 			update( cartItems.replaceItem( action.oldItem, action.newItem ) );
 			break;
+
+		case TRANSACTION_NEW_CREDIT_CARD_DETAILS_SET:
+			{
+				// typically set one or zero
+				const countryCode = get( action, 'rawDetails.country' );
+				const postalCode = get( action, 'rawDetails.postal-code' );
+				postalCode && update( setTaxPostalCode( postalCode ) );
+				countryCode && update( setTaxCountryCode( countryCode ) );
+			}
+			break;
+
+		case TRANSACTION_PAYMENT_SET:
+			{
+				// set both
+				let postalCode, countryCode;
+
+				const paymentMethod = get( action, [ 'payment', 'paymentMethod' ] );
+				switch ( paymentMethod ) {
+					case 'WPCOM_Billing_MoneyPress_Stored':
+						postalCode = extractStoredCardMetaValue( action, 'card_zip' );
+						countryCode = extractStoredCardMetaValue( action, 'country_code' );
+						break;
+					case 'WPCOM_Billing_MoneyPress_Paygate':
+						const paymentDetails = get( action, 'payment.newCardDetails', {} );
+						postalCode = paymentDetails[ 'postal-code' ];
+						countryCode = paymentDetails.country;
+						break;
+					default:
+						throw new Error( 'Unrecognized payment method', action );
```

Comment:
Sorry I missed this - there are other payment methods (like `WPCOM_Billing_WPCOM`) and this caused those payments to fail. (Reverted)

Reply:
Thanks - I'll fix add appropriate tests and try again